common/src/main/java/io/homeassistant/companion/android/common/data/prefs/impl/entities/TemplateTileConfig.kt

```diff
@@ -0,0 +1,36 @@
+package io.homeassistant.companion.android.common.data.prefs.impl.entities
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import org.json.JSONObject
+
+const val FIELD_TEMPLATE = "template"
+const val FIELD_REFRESH_INTERVAL = "refresh_interval"
+
+data class TemplateTileConfig(
+    @JsonProperty(value = FIELD_TEMPLATE)
+    val template: String,
+    @JsonProperty(value = FIELD_REFRESH_INTERVAL)
+    val refreshInterval: Int
+) {
+    constructor(jsonObject: JSONObject) : this(
+        jsonObject.getString(FIELD_TEMPLATE),
+        jsonObject.getInt(FIELD_REFRESH_INTERVAL)
+    )
+
+    fun toJSONObject(): JSONObject {
+        return JSONObject(
+            mapOf(
+                FIELD_TEMPLATE to template,
+                FIELD_REFRESH_INTERVAL to refreshInterval
+            )
+        )
+    }
+
+    fun withTemplate(template: String): TemplateTileConfig {
+        return TemplateTileConfig(template, refreshInterval)
+    }
+
+    fun withRefreshInterval(refreshInterval: Int): TemplateTileConfig {
+        return TemplateTileConfig(template, refreshInterval)
+    }
```

Comment:
Why create these functions when Kotlin data classes have the built-in [copy function for this purpose](https://kotlinlang.org/docs/data-classes.html#copying) (for example `templateTileConfig?.copy(template = newTemplate)`)?

Reply:
Probably because this is the first time I'm using Kotlin for real and I'm used to Java :rofl: Thanks, fixed (and removed the methods):

```diff
diff --git app/src/full/java/io/homeassistant/companion/android/settings/wear/SettingsWearViewModel.kt app/src/full/java/io/homeassistant/companion/android/settings/wear/SettingsWearViewModel.kt
index fd091489..2a26f85e 100644
--- app/src/full/java/io/homeassistant/companion/android/settings/wear/SettingsWearViewModel.kt
+++ app/src/full/java/io/homeassistant/companion/android/settings/wear/SettingsWearViewModel.kt
@@ -145,7 +145,7 @@ class SettingsWearViewModel @Inject constructor(
     fun setTemplateTileContent(tileId: Int, updatedTemplateTileContent: String) {
         val templateTileConfig = templateTiles[tileId]
         templateTileConfig?.let {
-            templateTiles[tileId] = it.withTemplate(updatedTemplateTileContent)
+            templateTiles[tileId] = it.copy(template = updatedTemplateTileContent)
             renderTemplate(tileId, updatedTemplateTileContent)
         }
     }
@@ -153,7 +153,7 @@ class SettingsWearViewModel @Inject constructor(
     fun setTemplateTileRefreshInterval(tileId: Int, refreshInterval: Int) {
         val templateTileConfig = templateTiles[tileId]
         templateTileConfig?.let {
-            templateTiles[tileId] = it.withRefreshInterval(refreshInterval)
+            templateTiles[tileId] = it.copy(refreshInterval = refreshInterval)
         }
     }

diff --git wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt
index 976d76db..9d471af1 100644
--- wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt
+++ wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt
@@ -497,7 +497,7 @@ class MainViewModel @Inject constructor(
         viewModelScope.launch {
             homePresenter.setTemplateTileRefreshInterval(tileId, interval)
             templateTiles[tileId]?.let {
-                templateTiles[tileId] = it.withRefreshInterval(interval)
+                templateTiles[tileId] = it.copy(refreshInterval = interval)
             }
         }
     }
```

Reply:
I think you forgot to push this commit

Reply:
Staged but not commited :see_no_evil: Pushed now.